homeassistant/components/loqed/lock.py

```diff
@@ -76,6 +76,11 @@ async def async_open(self, **kwargs: Any) -> None:
         """Open the door latch."""
         await self._lock.open()
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return custom state attributes."""
+        return {EXTRA_ATTRIBUTE_LAST_EVENT: self._lock.last_event}
```

Comment:
I think it would be better if this was separate sensor (with an enum), that supports translations to make it look better in the UI.

Reply:
My only concerns with these approach are following:
- there might be a potential inconsistency between separate Last event entity and Lock entity states, because of async nature of entity updates
- I don't feel confident enough that a new states won't be added/changed to the LOQED firmware later.

Reply:
> there might be a potential inconsistency between separate Last event entity and Lock entity states, because of async nature of entity updates

As this uses a coordinator, that argument will be the same for the suggested attribute in this PR and with the separate sensor.

> I don't feel confident enough that a new states won't be added/changed to the LOQED API/firmware later.

Then we will discover that, as an invalid value will raise and we can add the translation that is needed after that.

../Frenck

Reply:
Oke. Fair point. 👍

I'll move it to separate entity.

Thanks for feedback!

Reply:
Done. Please check again :)